helios-crtauth/src/main/java/com/spotify/helios/auth/crt/CrtAuthenticationPlugin.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2014 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.auth.crt;
+
+import com.google.auto.service.AutoService;
+import com.google.common.annotations.VisibleForTesting;
+
+import com.spotify.crtauth.CrtAuthServer;
+import com.spotify.crtauth.keyprovider.KeyProvider;
+import com.spotify.helios.auth.AuthenticationPlugin;
+
+import org.springframework.ldap.core.LdapTemplate;
+import org.springframework.ldap.core.support.LdapContextSource;
+
+import java.util.Map;
+
+@AutoService(AuthenticationPlugin.class)
+public class CrtAuthenticationPlugin implements AuthenticationPlugin<CrtAccessToken> {
+
+  private final Map<String, String> environment;
+
+  public CrtAuthenticationPlugin() {
+    this(System.getenv());
+  }
+
+  @VisibleForTesting
+  protected CrtAuthenticationPlugin(Map<String, String> environment) {
+    this.environment = environment;
+  }
+
+  @Override
+  public String schemeName() {
+    return "crtauth";
+  }
+
+  @Override
+  public ServerAuthentication<CrtAccessToken> serverAuthentication() {
+    // only validate the presence of environment variables when this method is called, as opposed to
+    // in the constructor, as the client-side code will not use the same environment variables
+    final String ldapUrl =  getRequiredEnv("CRTAUTH_LDAP_URL");
```

Comment:
Would requiring config file(s) be better than env vars? We could look in `/etc/default/helios-master` (which is already being written to and read from by the debian package) and then fallback to a json file in the current working directory.